packages/core/src/contrib/Constraints.ts

```diff
@@ -162,12 +168,10 @@ export const objDict = {
       );
 
       // entire equation is (mx - lx) ^ 2 + (my + 1.1 * text.h - ly) ^ 2 from Functions.hs - split it into two halves below for readability
-      const lh = squared(sub(mx, text.center.contents[0]));
```

Comment:
I would avoid using `bbox` here as well, since the text center and dimensions are available from the shape itself. In general, I would avoid unnecessary computation in an objective/constraint as it adds extra ops to the energy/gradient evaluation (which is the "hottest" path in the system).

Reply:
`h` is used later, so we already need to construct a `BBox`.

Reply:
In the new version of `BBox` in cases where the input is a non-Square rect-like and the only fields used are `w`, `h`, and `center`, the overhead of the BBox is just creating a new BBox object. The constraint graph is not made more complex.